p2p/src/protocol.rs

```diff
@@ -929,7 +929,10 @@ impl<T: BlockTree, F: Filters, P: peer::Store> Protocol<T, F, P> {
                 (*self.hooks.on_getdata)(addr, invs, &self.upstream);
             }
             _ => {
-                debug!(target: self.target, "{}: Ignoring {:?}", addr, cmd);
+                //debug!(target: self.target, "{}: Ignoring {:?}", addr, cmd);
+                // We receive a unknown message, send this information to the peer.
+                // FIXME(vincenzopalazzo): we can wrap this call inside the peermgr?
+                self.upstream.unknown_msg(addr, msg.command());
```

Comment:
:confused: 

Why would we echo back the message we received?

Reply:
I think this comes from a misunderstanding of "upstream": it is a channel to send messages to the network.

Reply:
>I think this comes from a misunderstanding of "upstream": it is a channel to send messages to the network.

It is totally my fault, it is not a misunderstanding of "upstream". I miss some lines of comment in the rust-bitcoin and make too many assumptions. Now it should look ok.

Thanks to catch my mistake.